arangod/Cluster/ClusterFeature.cpp

```diff
@@ -143,21 +143,41 @@ void ClusterFeature::validateOptions(std::shared_ptr<ProgramOptions> options) {
 
   // check if the cluster is enabled
   _enableCluster = !_agencyEndpoints.empty();
-
   if (!_enableCluster) {
     _requestedRole = ServerState::ROLE_SINGLE;
     ServerState::instance()->setRole(ServerState::ROLE_SINGLE);
-    auto ss = ServerState::instance();
-    ss->findHost("localhost");
+    ServerState::instance()->findHost("localhost");
     return;
   }
 
-  // validate --cluster.agency-endpoint (currently a noop)
+  // validate --cluster.agency-endpoint
   if (_agencyEndpoints.empty()) {
     LOG_TOPIC(FATAL, Logger::CLUSTER)
       << "must at least specify one endpoint in --cluster.agency-endpoint";
     FATAL_ERROR_EXIT();
   }
+  
+  // validate --cluster.my-address
+  if (_myEndpoint.empty()) {
+    LOG_TOPIC(FATAL, arangodb::Logger::CLUSTER) << "unable to determine internal address for server '"
+    << ServerState::instance()->getId()
+    << "'. Please specify --cluster.my-address or configure the "
+    "address for this server in the agency.";
```

Comment:
We should adjust this error message, since we no longer support to configure the address via the agency.